src/components/Header.tsx

```diff
@@ -0,0 +1,98 @@
+import {
+  useContext, useEffect, useRef,
+} from 'react';
+import { TodosContext } from '../context/TodoContext';
+
+export const Header = () => {
+  const {
+    todos,
+    setNewTodoTitle,
+    newTodoTitle,
+    handleError,
+    addTodo,
+    USER_ID,
+    disabledInput,
+    setDisabledInput,
+    // setTempTodo,
+    // tempTodo,
+    // setError,
+  } = useContext(TodosContext);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, []);
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setDisabledInput(true);
+
+    if (newTodoTitle.trim() === '') {
+      handleError('Title should not be empty');
+      setDisabledInput(false);
+
+      return;
+    }
+
+    const todo = {
+      title: newTodoTitle.trim(),
```

Comment:
if you will trim it above the newTodo (i suggest you to change the name of variable todo -> newTodo), so you can use here just newTodoTItle